### PR TITLE
feat: support flavored markdown in messages

### LINKED ISSUE #10066: Support Github Flavored Markdown
When https://github.com/nextcloud/nextcloud-vue/pull/4365 is there, messages styles should be updated to properly support:
  - ~autolink literals~
  - footnotes
  - ~strikethrough~
  - tables
  - tasklists

_Originally posted by @Antreesy in https://github.com/nextcloud/spreed/issues/10063#issuecomment-1653813378_

### LINKED ISSUE #11405: Strike text is not working on web ui `~~striked text~~`
When striking text on the WEB ui, it doesn't work.
Android works well though

| Web | Android |
|:---------:|:------:|
|image|image|

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Fix #10066 
* Fix #11405
* Styles are moved to scss file, to be reused by all required components
* Follow-up: interactive checkboxes (align with message editing)

### 🖼️ Screenshots / Screencasts

| 🏚️ Before | 🏡 After |
|------------|----------|
Screenshot from 2024-01-30 20-22-22 | Screenshot from 2024-01-30 20-09-00

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [ ] 🖌️ Design was reviewed, approved or inspired by the design team
- [ ] ⛑️ Tests are included or not possible
- [ ] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

### CODE REVIEW

File: src/components/ConversationSettings/EditableTextField.vue [hunk 1]
```diff
@@ -279,6 +279,11 @@ export default {
 	:deep(.rich-contenteditable) {
 		flex-grow: 1;
 	}
+
+	:deep(.rich-text--wrapper) {
+		text-align: start;
```
Comment (DorraJaouad) [inline src/components/ConversationSettings/EditableTextField.vue:284]: Why `text-align: start;` ?

**Review by DorraJaouad — APPROVED** [review]
Thanks! Nothing blocking.

File: src/components/ConversationSettings/EditableTextField.vue (hunk 1 above)
Comment (Antreesy) [reply src/components/ConversationSettings/EditableTextField.vue:284]: For right-to-left languages

Comment (Antreesy) [reply src/components/ConversationSettings/EditableTextField.vue:287]: Updated styles for tables to not overflow the message body, and updated template for RTL support. Good to go

**Pull request merged**